registry/src/types.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proptest::prelude::*;
+
+use std::{ops::Deref, str::FromStr, sync::Arc};
+use test_strategy::Arbitrary;
+
+prop_compose! {
+    fn package_name()(name in "[a-z][a-z0-9-]{0,127}") -> Arc<str> {
+        name.to_string().into()
+    }
+}
+
+prop_compose! {
+    fn package_version()(major: u16, minor: u16, patch: u16) -> Arc<str> {
+        format!("{major}.{minor}.{patch}").into()
+    }
+}
+
+prop_compose! {
+    fn token()(token in "[a-z]{64}") -> Arc<str> {
+        token.to_string().into()
+    }
+}
+
+prop_compose! {
```

Comment:
This doesn't seem like it should live here

Reply:
Removed it!